osu.Game/Screens/SelectV2/BeatmapCarouselFilterGrouping.cs

```diff
@@ -152,12 +160,15 @@ private List<GroupMapping> getGroups(List<CarouselItem> items, FilterCriteria cr
                 case GroupMode.LastPlayed:
                     return getGroupsBy(b =>
                     {
-                        DateTimeOffset? maxLastPlayed = aggregateMax(b, items, bb => bb.LastPlayed);
+                        var date = b.LastPlayed;
 
-                        if (maxLastPlayed == null)
+                        if (BeatmapSetsGroupedTogether)
+                            date = aggregateMax(b, static b => (b.LastPlayed ?? DateTimeOffset.MinValue));
```

Comment:
Pretty sure this isn't gonna be right if all of the beatmaps have null last played (will fall back to `DateTimeOffset.MinValue` rather than null and probably show something stupid)

Reply:
Turns out this was actually the reason for the test failures 🤦